src/libraries/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/IsolatedStorageFile.NonMobile.cs

```diff
@@ -0,0 +1,19 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text;
+
+namespace System.IO.IsolatedStorage
+{
+    public sealed partial class IsolatedStorageFile : IsolatedStorage, IDisposable
+    {
+        private string GetIsolatedStorageRoot()
+        {
+            StringBuilder root = new StringBuilder(Helper.GetRootDirectory(Scope));
+            root.Append(SeparatorExternal);
+            root.Append(IdentityHash);
```

Comment:
For mobile platforms main issue was that projects that were released on legacy Xamarin.Android will lose access to files created with IsolatedStorageFile after upgrading the project to .NET 6. So even if hash was consistent it would not solve the issue. I checked only for mobile platforms and it was changing only on iOS, android was consistent for different builds.

Reply:
Right, I'm wondering if the IdentityHash changing on iOS is intended or not. If its a bug and iOS should have a non-changing IdentityHash, then we wouldn't have to branch out this file.

Reply:
We need to branch out as on legacy Xamarin.iOS path was /var/mobile/Containers/Data/Application/<GUID>/Documents/.config/.isolated-storage/ without including any IdentityHash. So solution is removing IdentityHas on mobile platforms